Add extra volumes to server Helm chart (#12108)
uyuni-project/uyuni: containers/server-helm/README.md

The storage class supports a special `"-"` value to force the storage class to the empty string.
This is different from the `""` value representing the default storage class by not setting any storage class at all.

Additional volumes can be attached to the server pod with `server.extraVolumes` and mounted into the server container with `server.extraVolumeMounts`.

> This is useful for optional data sources such as benchmark repositories or externally managed content volumes without changing the chart templates.

Reviewer: ```suggestion
This is useful for optional content volumes like `inter-server-sync` data without changing the chart templates.
```
I know the original use case is benchmark data, but users won't have that case. A mounted volume to export the data in order to reload them on another server will make more sense to users.

For example:

```yaml

> server:
>   extraVolumes:
>     - name: uyuni-bench-source
>       persistentVolumeClaim:
>         claimName: uyuni-bench-source
> 
>   extraVolumeMounts:
>     - name: uyuni-bench-source
>       mountPath: /uyuni-bench-source
>       readOnly: true

Reviewer: ```suggestion
server:
  extraVolumes:
    - name: inter-server-sync-data
      persistentVolumeClaim:
        claimName: inter-server-sync-data

  extraVolumeMounts:
    - name: inter-server-sync-data
      mountPath: /inter-server-sync-data
      readOnly: true
```

This example could almost be used as is by some users.
